ERCS/erc-7828.md

```diff
@@ -1,80 +1,86 @@
 
 Target address
 : The (address, chain ID) pair a particular Interoperable Address points to.
 
-Name-resolving contract
-: A contract responsible for converting addresses and/or chain IDs from a machine-readable format into parts of a human-readable name.
-
-## Out of scope concerns
-Similarly to CAIP-10, this specification is not concerned with the mapping from a chain ID to a network name, which might not be surjective (eg: the case where if there are multiple EIP-155 chains with chain ID 8453, which one should we call Base?), regarding that resolution a social-layer problem until a future ERC decides to tackle it. Efforts in that front are tracked in [the chain registries document](../docs/chain-registries.md)
-
 ## Guarantees provided by the standard
-- Any Interoperable Address is trivially convertible to Interoperable Canonical Format (specified below), making it a canonical representation for users who need to treat them opaquely.
-- Checksums are short enough to be compared by humans, and effectively identify a _target address_ independently of any extra data the address may contain.
+- Any future Interoperable Address will be trivially convertible to Interoperable Address v1 (specified below), which, together with the point below, makes them a canonical representation for users who need to treat them opaquely.
+- Two Interoperable Addresses, converted to this canonical version, will be bitwise-equal iff they represent the same _target address_.
+- Checksums are short enough to be compared by humans, and allow for easy differentiation of _target addresses_ independently of any extra data the Interoperable Address may contain.
 
 ## General specification
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
-### Binary format definition
-Word 0 Is the only word which will be common to all versions, and is defined as follows:
+### Interoperable Address binary format definition
+In binary format addresses have the following encoding:
+
 ```
-MSB                                                            LSB
-0xXXXXYYYYYYYYZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ
-  ^^^^------------------------ 255-240 Interoperable Address version
-      ^^^^^^^^ --------------- 239-208 Checksum
-              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-                \------------- 207-0   Reserved
+┌─────────┬────────────┬─────────┬─────────┬─────────┐
+│ Version │ Chainidlen │ Chainid │ Addrlen │ Address │
+└─────────┴────────────┴─────────┴─────────┴─────────┘
 ```
 
-No length restriction is placed on how many more words can an Interoperable Address span. Different versions of the standard can define uses for extra words and the 208 reserved bits.
+Where:
+
+Version
+: a 2-byte version identifier. In the current standard this must always be 1.
```

Comment:
I don't think there's any reason to make this 2 bytes; why not follow a similar process to UTF-8: Specify that all versions presently defined must leave the MSB unset, leaving open the opportunity to later extend the standard with multi-byte version IDs.